src/query.rs

```diff
@@ -65,6 +65,10 @@ pub unsafe trait Fetch<'a>: Sized {
     unsafe fn get(&self, n: usize) -> Self::Item;
 }
 
+/// Marker trait indicating whether a given [`Fetch`] implementation will produce only shared references
+#[allow(clippy::missing_safety_doc)]
+pub unsafe trait FetchShared {}
```

Comment:
Could we mark the query types themselves, rather than the `Fetch` impls? That should be a bit more intuitive for users.

Reply:
Changed.